### ISSUE TITLE
heading value display bug

### ISSUE THREAD

**Issue by gerrygk** [pioneerspacesim/pioneer#3550]
Hey guys,

I've found an incorrect heading value display in v 2015-11-09. This value counts up to 360 instead of 359 before it continues with 0.

Greetings

**Comment by impaktor** [comment]
#3531 fixed this I think.

But thanks for reporting.

**Comment by gerrygk** [comment]
Hi,

as seen in v 2015-11-30 obviously not. Heading degrees 360 and 0 are 
both still appearing.

Greetings

Am 01.12.2015 um 16:01 schrieb Karl F:

**Comment by impaktor** [comment]
OK, I thought you were reporting another very similar bug.

So straight ahead is either 0 or 360 degrees depending on which way you've rotated there? It's not obvious to me that this is a bug, but: ping @clausimu.

**Comment by gerrygk** [comment]
No, the counting doesn't depend on turning left or right. I always have 0 after the 360 or vice versa.

----- Ursprüngliche Nachricht -----
Von: "Karl F" <notifications@github.com>
Gesendet: ‎05.‎12.‎2015 10:07
An: "pioneerspacesim/pioneer" <pioneer@noreply.github.com>
Cc: "gerrygk" <gerryk70@web.de>
Betreff: Re: [pioneer] heading value display bug (#3550)

OK, I thought you were reporting another very similar bug.

So straight ahead is either 0 or 360 degrees depending on which way you've rotated there? It's not obvious to me that this is a bug, but: ping @clausimu.

---

**Comment by johnbartholomew** [comment]
Headings in the range 359.5 to 360 would cause this. It could probably be fixed by wrapping 359.5 to -0.5.

**Comment by gerrygk** [comment]
Maybe, but that's the programmer's task. 😉

----- Ursprüngliche Nachricht -----
Von: "John Bartholomew" <notifications@github.com>
Gesendet: ‎05.‎12.‎2015 15:45
An: "pioneerspacesim/pioneer" <pioneer@noreply.github.com>
Cc: "gerrygk" <gerryk70@web.de>
Betreff: Re: [pioneer] heading value display bug (#3550)

Headings in the range 359.5 to 360 would cause this. It could probably be fixed by wrapping 359.5 to -0.5.

---

**Comment by radius75** [comment]
it seems to me that a solution must be similar to this:

```ruby
@@ -2257,8 +2257,14 @@ static std::pair<double, double> calculateHeadingPitch(PlaneType pt) {
 	const double pitch = asin(up.Dot(hed));
 
 	const double hedNorth = groundHed.Dot(north);
 	const double hedEast = groundHed.Dot(east);
 	const double heading = wrapAngleToPositive(atan2(hedEast, hedNorth));
+	if (wrapAngleToPositive(atan2(hedEast, hedNorth)) > (359.4 * 2 * M_PI / 360))
+
+		(const double (heading)) = 0;
+
+
+	
 
 	return std::make_pair(heading, pitch);
 }

```
